### PR TITLE
Rename data input keys

### PR DESCRIPTION
**Pull request by Eleonore9**
- Name the population entering the core component `:base-popn`.
- Update the schema name.
- Remove the step to add a `:year` to the data input map.

=> more changes to be addressed on the next branch

### CODE REVIEW

File: src/witan/models/dem/ccm/core/projection_loop.clj [hunk 1]
```diff
@@ -25,23 +25,27 @@
...
-(defworkflowfn core-loop
+(defworkflowfn ccm-core
   "Takes in a dataset of popn estimates, a first year
    and last year. Implement the looping to output the projections
    for the range of years between the first and last year."
-  {:witan/name :exec-core-loop
+  {:witan/name :exec-core-ccm
    :witan/version "1.0"
-   :witan/input-schema {:historic-popn-estimates HistoricPopnEstimates}
+   :witan/input-schema {:base-popn HistoricPopnEstimates}
```
Comment (acron0) [inline src/witan/models/dem/ccm/core/projection_loop.clj]: I think this schema `HistoricPopnEstimates` wants a new name too.

Comment (Eleonore9) [inline src/witan/models/dem/ccm/core/projection_loop.clj]: Yep, good point!

File: src/witan/models/load_data.clj [hunk 2]
```diff
@@ -45,7 +45,7 @@
   (make-ordered-ds-schema [[:gss-code s/Str] [:district s/Str] [:sex s/Str] [:age s/Int]
                            [:var s/Str] [:year s/Int] [:estimate s/Num]]))
 
-(def HistPopnEst
+(def BasePopn
```
Comment (acron0) [inline src/witan/models/load_data.clj]: Is it necessary to repeat this schema here and https://github.com/MastodonC/witan.models.demography/pull/18/files#diff-e7253d5ba9334b4ad5edd60a2aa084b7R14 ?

Comment (Eleonore9) [inline src/witan/models/load_data.clj]: I'm not sure. Load-data is used by the test ns so far to apply coercion on the uploaded testing data.
Ultimately we'll have several code ns w/ schemas. Do we want to require those ns in the load-data ns?

Comment (acron0) [inline src/witan/models/load_data.clj]: I think the less repetition of code, the better. Schemas are best in their own NS, usually

File: src/witan/models/dem/ccm/core/projection_loop.clj [hunk 3]
```diff
@@ -11,37 +11,37 @@
...
-   :witan/param-schema {:year s/Int}
-   :witan/output-schema {:starting-popn StartingPopn}}
-  [{:keys [historic-popn-estimates]} {:keys [year]}]
-  {:starting-popn (i/query-dataset historic-popn-estimates {:year (dec year)})})
+   :witan/input-schema {:popn PopnSchema}
+   :witan/param-schema {:first-proj-year s/Int}
+   :witan/output-schema {:starting-popn PopnSchema}}
+  [{:keys [popn]} {:keys [first-proj-year]}]
+  (let [latest-yr (reduce max (i/$ :year popn))]
+    (if (>= latest-yr first-proj-year)
```
Comment (acron0) [inline src/witan/models/dem/ccm/core/projection_loop.clj:29]: You don't want to handle this logic here. There's no good story for throwing exceptions inside workflow fns but they absolutely should be avoided. The loop logic will be in a predicate, outside of the workflow fns so I'd move it there and have it in your `ccm-core` function, not inside your `starting-popn` function. Still not even sure we need this function...

Comment (Eleonore9) [inline src/witan/models/dem/ccm/core/projection_loop.clj:29]: Not sure we need the `ccm-core` function either but keeping it for now. Will remove it when I know it's useless.

Thanks, as I said during demo yesterday I wasn't sure it was the place for the exception throwing. I can definitely have this check outside of this function.

**Comment by sunnytownsend** [comment]
:shipit:

**Pull request merged**